lua/nui/popup/popup.md

```diff
@@ -0,0 +1,460 @@
+    highlight = "MyHighlightGroup",
+    padding = {
+      top = 2,
+        bottom = 2,
+        left = 3,
+        right = 3,
+    },
+    style = "double",
+    text = {
+      top = " I am top title ",
+      top_align = "center",
+      bottom = "I am bottom title",
+      bottom_align = "left",
+    },
+  },
+  buf_options = {
+    modifiable = true,
+    readonly = false,
+  },
+  win_options = {
+    winblend = 10,
+    winhighlight = "Normal:Normal",
+  },
+)
+```
+
+## Property Descriptions
+
+### `border`
+
+- **Type:** `table`
+- Contains all border related options
+
+### `border.padding`
+
+- **Type:** `table`
+- Controls the popup padding
+- Behaves like [CSS padding](https://developer.mozilla.org/en-US/docs/Web/CSS/padding)
+
+**Examples**
+
+```lua
+border = {
+  padding = { top = 10, bottom = 20, left = 15, right = 20 }
+}
+```
+
+```lua
+border = {
+  padding = { 10, 30, 40, 10 }
+}
+```
+
+### `border.highlight`
+
+- **Type:** `string`
+- Highlight group name for the border characters.
+
+```lua
+border = {
+  highlight = "MyHighlightGroup"
+}
+```
+
+### `border.style`
+
+- **Type:** `string` or `table`
+- Controls the styling of the border
+
+**Examples**
+
+- Can be: `double`, `none`, `rounded`, `shadow`, `single`, `solid`
+
+```lua
+border = {
+  style = "double"
+}
+```
+
+- Order is clockwise
+
+```lua
+border = {
+  style = { "╭", "─", "╮", "│", "╯", "─", "╰", "│" }
+}
+```
+
+```lua
+border = {
+  style = {
+    top_left = "╭",
+    top = "─",
+    top_right = "╮",
+    left = "│",
+    right = "│",
+    bottom_left = "╰",
+    bottom = "─",
+    bottom_right = "╯",
+  }
+}
+```
+
+#### `border.text`
+
+- **Type:** `string` or `table`
+- Text displayed in the border to serve as a title
+- `top_align` and `bottom_align` can be: `left`, `right`, `center`
+
+**Examples**
+
+```lua
+border = {
+  text = {
+    top = "I am top title",
+    top_align = "center",
+    bottom = "I am bottom title",
+    bottom_align = "left",
+  }
+}
+```
+
+---
+
+### `relative`
+
+- **Type:** `string` or `table`
+- This option affects how `position` and `size` is calculated.-
+
+**Examples**
+
+- Relative to cursor on current window
+
+```lua
+relative = "cursor"
+```
+
+- Relative to the current editor on current window
+
+```lua
+relative = "editor"
+```
+
+- Relative to the current window
+
+```lua
+relative = "win"
+```
+
+- Relative to the window of `id`
+
+```lua
+relative = {
+  type = "win",
+  winid = 5,
+}
+```
+
+- Relative to the buffer's `position`
+
+```lua
+relative = {
+  type = "buf",
+  position = {
+    row = 5,
+    col = 5,
+  },
+}
+```
+
+---
+
+#### `position`
+
+- **Type:** `number` or `percentage string` or `table`
+- If `position` is `number` or `percentage string`, it applies to both `row` and `col`
+- Position is calculated from the top-left corner
+- For `percentage`, position is calculated according to the option `relative`
+  - If `relative` is set to `"buf"` or `"cursor"`, `percentage string` is not allowed
```

Comment:
Do you mean for all of them or just his particular one? 

I find bullet points a lot easier to read as its just bite-sized statements instead of bigger paragraphs.

But again, up to you.

Reply:
> Do you mean for all of them or just his particular one?

all of them

> I find bullet points a lot easier to read as its just bite-sized statements instead of bigger paragraphs.

You got a point there. But these information are not really suitable for lists... seeing them in list seems kinda forced. 🤔 Also the paragraphs doesn't need to be big. You can have bite-sized paragraphs as well. 😃